### ISSUE TITLE
Test fail when no credentials are provided

### ISSUE THREAD

**Issue by LaurinFischer** [Qiskit-Extensions/qiskit-cold-atom#10]
Opening an issue here due to some strange behaviour that Fred described here: #8 

The tests seemingly fail as no Heidelberg provider credentials were provided:
```
% python -m unittest
/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py:289: UserWarning: No credentials found at the specified location
  warnings.warn("No credentials found at the specified location").
```

**Comment by fretchen** [comment]
Here is the error output with six failing tests:

```
python -m unittest
.../Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py:289: UserWarning: No credentials found at the specified location
  warnings.warn("No credentials found at the specified location")
EEE./Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/fermions/fermion_gate_library.py:328: UserWarning: The default value for `display_format` will be changed from 'dense' to 'sparse' in version 0.3.0. Once that happens, you must specify display_format='dense' directly.
  coeff * FermionicOp(label, register_length=self.num_modes)
......................../opt/miniconda3/envs/qiskit-cold-atom/lib/python3.9/site-packages/scipy/sparse/_index.py:82: SparseEfficiencyWarning: Changing the sparsity structure of a csc_matrix is expensive. lil_matrix is more efficient.
  self._set_intXint(row, col, x.flat[0])
.........................
======================================================================
ERROR: test_credential_management (test.test_cold_atom_provider.TestHeidelbergProvider) [test save account]
Test the management of locally stored credential data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 36, in test_credential_management
    ColdAtomProvider.save_account(
  File "/Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py", line 206, in save_account
    with open(filename, "w") as credentials_file:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fnj/.qiskit/cold_atom_provider_test'

======================================================================
ERROR: test_credential_management (test.test_cold_atom_provider.TestHeidelbergProvider) [test overwrite warning]
Test the management of locally stored credential data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 52, in test_credential_management
    ColdAtomProvider.save_account(
  File "/Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py", line 206, in save_account
    with open(filename, "w") as credentials_file:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fnj/.qiskit/cold_atom_provider_test'

======================================================================
ERROR: test_credential_management (test.test_cold_atom_provider.TestHeidelbergProvider) [test add url]
Test the management of locally stored credential data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py", line 228, in add_url
    urls = credentials_dict["urls"]
KeyError: 'urls'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 60, in test_credential_management
    ColdAtomProvider.add_url("second_url", filename=self.filename)
  File "/Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py", line 232, in add_url
    raise QiskitColdAtomError("No stored credentials found") from err
qiskit_cold_atom.exceptions.QiskitColdAtomError: 'No stored credentials found'

======================================================================
ERROR: test_credential_management (test.test_cold_atom_provider.TestHeidelbergProvider)
Test the management of locally stored credential data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 101, in tearDown
    os.remove(self.filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fnj/.qiskit/cold_atom_provider_test'

======================================================================
ERROR: test_provider_initialization (test.test_cold_atom_provider.TestHeidelbergProvider)
Test the initialization of a cold atom provider from invalid credentials
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 82, in test_provider_initialization
    ColdAtomProvider.save_account(
  File "/Users/fnj/gitKIP/qiskit-cold-atom/qiskit_cold_atom/providers/cold_atom_provider.py", line 206, in save_account
    with open(filename, "w") as credentials_file:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fnj/.qiskit/cold_atom_provider_test'

======================================================================
ERROR: test_provider_initialization (test.test_cold_atom_provider.TestHeidelbergProvider)
Test the initialization of a cold atom provider from invalid credentials
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/fnj/gitKIP/qiskit-cold-atom/test/test_cold_atom_provider.py", line 101, in tearDown
    os.remove(self.filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fnj/.qiskit/cold_atom_provider_test'

----------------------------------------------------------------------
Ran 55 tests in 2.297s

FAILED (errors=6)
```

**Comment by LaurinFischer** [comment]
Thanks Fred, 
this bug is addressed in #11 . It only appeared for users who don't have IBMQ credentials saved in the default location which is why we didn't catch it earlier.

**Comment by eggerdj** [comment]
Solved by #11
